cluster/gce/util.sh

```diff
@@ -912,6 +912,16 @@ function detect-subnetworks() {
   echo "${color_red}Could not find subnetwork with region ${REGION}, network ${NETWORK}, and project ${NETWORK_PROJECT}"
 }
 
+function delete-all-firewall-rules-in-network() {
```

Comment:
I would suggest just adding the following code inside `delete-network` where currently we're just listing the firewall-rules and returning in case the network failed to delete.

Just need to add an if statement there like this:

```
if fws=$(gcloud compute firewall-rules list --project "${NETWORK_PROJECT}" --filter="network=${NETWORK}" --format="value(name)"); then
  echo "Deleting firewall rules remaining in network ${NETWORK}: ${fws}"
  delete-firewall-rules "$fws"
else
  echo "Failed to list firewall rules from the network ${NETWORK}"
fi
```

Reply:
Thanks for the suggestion, seems much better and applied. Though I still put it as a separate function, it seems more reasonable if `detect-network()` only deletes network and cleanup of other resources happens in a separate function (like `delete-subnetworks()`).